### PR TITLE
Adding Jax in practice docs

### PR DESCRIPTION
**Pull request by canyon289**
Currently WIP. Pushing for discussion with Jake and Olzhaus

### CODE REVIEW

File: docs/jax201.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+or [Neural Network with JAX](https://jax.readthedocs.io/en/latest/notebooks/neural_network_with_tfds_data.html)
+
+### Model specific Libraries with JAX exposed
+Another common approach is to create a package that provides clean APIs for prepackaged models. An example of this architecture is [Dynamax](https://github.com/probml/dynamax),
+where state space models are already defined and users do some light specification and plug in their own values,
+or [Flax](https://github.com/google/flax) which simplifies the construction Neural Networks.
+
+### Composable Functionality
+Libraries within the JAX ecosystem are often mixed and matched.
+For example [Flax](https://github.com/google/flax) and [Optax](https://github.com/deepmind/optax) are commonly used together, where Flax defines the neural network architecture.
```
Comment (jakevdp) [inline docs/jax201.md:26]: "... and Optax extends its optimization & model-fitting capabilities"?

**Review by jakevdp — COMMENTED** [review]
It's a good start – still feels somewhat incomplete. Let's brainstorm what other examples we could mention

**Comment by canyon289** [comment]
Agree on it being incomplete. I'll keep adding to it. Good to know its heading in the right direction

**Comment by canyon289** [comment]
@jakevdp @8bitmp3 This is ready for another review. Curious where you two think I should take this next.

@8bitmp3 I used markdown as you suggested but its making cross linking, such as vmap and pmap, a bit challenging. If we'd like to do that I think I need to switch .rst

File: docs/jax201.md (hunk 1 above)
Comment (canyon289) [reply docs/jax201.md]: Added!

File: docs/jax201.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Jax in Practice
```
Comment (canyon289) [inline docs/jax201.md:1]:
```suggestion
# Building on JAX
```

File: docs/index.rst [hunk 3]
```diff
@@ -98,6 +98,7 @@ installation options, see the `Install Guide`_ in the project README.
    :maxdepth: 2
    :caption: Further Resources
 
+   jax201
```
Comment (canyon289) [inline docs/index.rst:101]: building_on_jax

Comment (canyon289) [reply docs/index.rst:101]: Change filename too

Comment (canyon289) [reply docs/index.rst:101]: Move to below, contributor_guide

**Review by jakevdp — APPROVED** [review]
Thanks!

**Pull request merged**